todo_app/app.py

```diff
@@ -1,41 +1,116 @@
+import functools
 from flask import Flask, render_template, request, redirect
-from todo_app.todo import Item
 from todo_app.flask_config import Config
 from todo_app.view_model import ViewModel
 from todo_app.todomongo import add_todo_item, items, update_status, delete_item
+import requests
+import os
+from flask_login import LoginManager, UserMixin, login_required, login_user, current_user
+
+class User (UserMixin):
+    def __init__(self, id):
+        self.id = id
+        self.is_reader = True
+        if id == '97612224':
+            self.roles=['reader', 'writer']
+        else:
+            self.roles=['reader']
 
 def create_app():
+    
     app = Flask(__name__)
     app.config.from_object(Config())
+    app.config['LOGIN_DISABLED'] = os.getenv('LOGIN_DISABLED') == 'True'
 
+    login_manager = LoginManager()
+    
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        redirect_url= f"https://github.com/login/oauth/authorize?client_id={os.getenv('GITHUB_CLIENT_ID')}"
+        return redirect(redirect_url)
+   
+    @login_manager.user_loader
+    def load_user(user_id):
+        return User(user_id)
 
-    @app.route('/')
+    
+    login_manager.init_app(app)
+
+    def writer_required(func):
+        @functools.wraps(func)
+        def forbidden_if_not_writer_func(*args, **kwargs):
+            if os.getenv('LOGIN_DISABLED') or 'writer' in current_user.roles:
```

Comment:
I can see you've adjusted it elsewhere, but this might be nicer updated here too to:
```suggestion
            if os.getenv('LOGIN_DISABLED') == 'True' or 'writer' in current_user.roles:
```
since otherwise setting it to "False" might cause you headaches!